WordPressAuthenticator/Extensions/UIPasteboard+Detect.swift

```diff
@@ -25,4 +25,50 @@ extension UIPasteboard {
             }
         }
     }
+
+    /// Attempts to detect and return a authenticator code from the pasteboard.
+    /// Expects to run on main thread.
+    /// - Parameters:
+    ///   - completion: Called with a six digit authentication code on success
+    @available(iOS 14.0, *)
+    public func detectAuthenticatorCode(completion: @escaping (Result<String, Error>) -> Void) {
+        UIPasteboard.general.detect(patterns: [.number]) { result in
+            switch result {
+                case .success(let detections):
+                    guard let firstMatch = detections.first else {
+                        completion(.success(""))
+                        return
+                    }
+                    guard let matchedNumber = firstMatch.value as? NSNumber else {
+                        completion(.success(""))
+                        return
+                    }
+
+                    var authenticationCode = matchedNumber.stringValue
+
+                    /// Reject numbers with decimal points or signs in them
+                    let codeCharacterSet = CharacterSet(charactersIn: authenticationCode)
+                    if !codeCharacterSet.isSubset(of: CharacterSet.decimalDigits) {
+                        completion(.success(""))
+                        return
+                    }
+
+                    /// We need 6 digits. No more, no less.
+                    if authenticationCode.count > 6 {
+                        completion(.success(""))
+                        return
+                    }
+
+                    while authenticationCode.count < 6 {
+                        authenticationCode = "0" + authenticationCode
+                    }
```

Comment:
This may be my crazy fears of avoiding `while` loops and mutable state, but would it make more sense to just prepend the necessary zeros here and avoid the loop?

```suggestion
                    let missingDigits = 6 - authenticationCode.count
                    let paddingZeros = String(repeating: "0", count: missing)
                    let paddedAuthenticationCode = paddingZeros + authenticationCode
```

This would also remove the need to make `authenticationCode` mutable, which I'm always in favor of.

Reply:
Fixed in https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/561/commits/e109be8783aa27c3b3bba97ee45ad559aaee4447